pkg/telemetry/events.go

```diff
@@ -145,6 +145,33 @@ func (t *telemetryService) ParticipantResumed(
 	reason livekit.ReconnectReason,
 ) {
 	t.enqueue(func() {
+		// create a worker if needed.
+		//
+		// Signalling channel stats collector and media channel stats collector could both call
+		// ParticipantJoined and ParticipantLeft.
+		//
+		// On a resume, the signalling channel collector would call `ParticipantLeft` which would close
+		// the corresponding participant's stats worker.
+		//
+		// So, on a successful resume, create the worker if needed.
+		worker, found := t.getOrCreateWorker(
+			ctx,
+			livekit.RoomID(room.Sid),
+			livekit.RoomName(room.Name),
+			livekit.ParticipantID(participant.Sid),
+			livekit.ParticipantIdentity(participant.Identity),
+		)
+		if !found {
+			// a worker was created, check if there is a closed worker and transfer connected state to new worker
+			if closedWorker := t.getClosedWorker(livekit.ParticipantID(participant.Sid)); closedWorker != nil {
+				if closedWorker.IsConnected() {
+					worker.SetConnected()
```

Comment:
This is a bit iffy. But, given that a worker is kept around for three minutes after `Close()` is called on it, in practice this should do the transfer of that state properly.